src/test/operators/join_index_test.cpp

```diff
@@ -102,14 +104,28 @@ class JoinIndexTest : public BaseTest {
     join->execute();
 
     EXPECT_TABLE_EQ_UNORDERED(join->get_output(), expected_result);
+
+    std::shared_ptr<const AbstractOperator> index_side_input;
+    if (index_side == IndexSide::Left) {
+      index_side_input = left;
+    } else {
+      index_side_input = right;
+    }
+
     const auto& performance_data = static_cast<const JoinIndex::PerformanceData&>(join->performance_data());
-    if (using_index && right->get_output()->type() == TableType::Data) {
-      // We can't execute the index join on referencing tables
-      EXPECT_EQ(performance_data.chunks_scanned_with_index, static_cast<size_t>(right->get_output()->chunk_count()));
+    // for index reference joins, only the join JoinMode::Inner is supported. Additionally, if the join segments of the
+    // reference table don't provde the guarantee of referencing one single chunk (of the original data table), then the
```

Comment:
(1) typo
(2) Instead of doc'ing in the test, I'd put it in the hpp. Especially the limitation to inner joins 
(3) Is there a valid case in which we would only realize at runtime that the scan is 100% fallback? If not, there should at least be a performance warning.
(4) On a similar note, if we end up with a non-inner join on a reference input, didn't someone else before screw up? Supporting this makes the code more complicated and introduces additional potential for errors.

Reply:
(4) : since any outer index join on a reference input is effectively a nested loop join, we should not support this setup here, right.

Reply:
Removing the setup "index side table is a ref. table and the join mode is non-inner" would require the static support function to be extended since we don't pass table data information.

Is there anything against this extension?

Current support function:
```c++
static bool supports(JoinMode join_mode, PredicateCondition predicate_condition, DataType left_data_type,
                       DataType right_data_type, bool secondary_predicates);
```

Reply:
Feel free to pass `TableType`